app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -94,7 +103,7 @@ def self.entrypoint(host, port, security_protocol)
     end
 
     # Hawkular Client
-    def self.raw_connect(host, port, username, password, security_protocol, cert_store)
+    def self.raw_connect(host, port, username, password, security_protocol, cert_store, validate = false)
```

Comment:
@israel-hdez do you know if MiQ uses `raw_connect`? If not I think this should be on a different class, probably a delegator. What do you think, @tumido?

If so, it could be changed on a different PR, because it seems to me that adding optional params is increasing cyclomatic complexity on the class that is already our biggest offender on this.

Reply:
Yes, they are using it :(

Reply:
@cfcosta, basically, the whole new workflow is about shifting the load from UI worker to spawning a queued job (created [here](https://github.com/ManageIQ/manageiq/blob/aa4dab9c75942a2c335d939a5eb112e3d25673df/app/models/mixins/authentication_mixin.rb#L16)) Also it tries to smaller the load by not creating a temporary EMS object for the credentials validation and instead validates them as is. So it requires a class method. And they decided, that using the already available `raw_connect` is the right way. For some providers it was already doing the job without any modifications, for others, like Hawkular, it means these adjustments...